orttraining/orttraining/core/graph/pipeline_transformer.cc

```diff
@@ -869,89 +869,217 @@ common::Status HandleSharedInitializer(Graph& graph,
   return Status::OK();
 }
 
-// split the graph into disconnected subgraph based on provided CutInfo
-common::Status SplitGraph(Graph& graph,
-                          std::vector<TrainingSession::TrainingConfiguration::CutInfo> split_edge_groups,
-                          std::vector<Node*>& send_nodes,
-                          std::vector<Node*>& recv_nodes) {
+// Returns all the pointers to NodeArg in the graph, before applying any 
+// partition transformation. 
+std::set<const NodeArg*> GetAllNodeArgs(Graph& graph) {
+  std::set<const NodeArg*> initial_node_args;
+  for (size_t i = 0, t = graph.MaxNodeIndex(); i < t; ++i) {
```

Comment:
Internally we had to change this loop to avoid iterating over invalid nodes: https://dev.azure.com/msrcambridge/Pylos/_git/onnxruntime/commit/f06ef28278d9a51a03581f65eb9883bc04175f87?refName=refs%2Fheads%2Fmsrc

Reply:
Yes, I need to fix this in a few places. My next push should take care of it. Thanks for the reminder.